ocfvirt.c

```diff
@@ -93,24 +98,19 @@ static int ocfvirt_response(Operation *op, SlapReply *rs) {
     if (!uid_bv) {
         return SLAP_CB_CONTINUE;
     }
-    const int ocfemail_len = email_suffix_len + (uid_bv ? uid_bv->bv_len : 0);
-
-    char *ocfemail_virt = ch_malloc(ocfemail_len + 1);
-    ocfemail_virt[0] = '\0';
 
-    if (uid_bv) {
-        strncat(ocfemail_virt, uid_bv->bv_val, uid_bv->bv_len);
-    }
-    strncat(ocfemail_virt, email_suffix, email_suffix_len);
+    assert(uid_bv->bv_len <= USERNAME_MAX);
 
     struct berval ocfemail_bv;
-    (void)ber_str2bv(ocfemail_virt, ocfemail_len, 0, &ocfemail_bv);
+    strncpy(ocfemail_virt, uid_bv->bv_val, USERNAME_MAX + 1);
```

Comment:
I believe is that `strncat` previously used is guaranteed to null-terminate dest, but not `strncpy`, and [`bv_val` isn't guaranteed to be null-terminated](https://www.openldap.org/software//man.cgi?query=ber_free&sektion=3&apropos=0&manpath=OpenLDAP+2.4-Release). This might only a problem if `bv_len == USERNAME_MAX`. You should double check this though.

Reply:
ugh thanks. fucking ldap lol